lib/std/tempfiles.nim

```diff
@@ -103,46 +116,60 @@ proc genTempPath*(prefix, suffix: string, dir = ""): string =
   let dir = getTempDirImpl(dir)
   result = dir / (prefix & randomPathName(nimTempPathLength) & suffix)
 
-proc createTempFile*(prefix, suffix: string, dir = ""): tuple[fd: File, path: string] =
-  ## Creates a new temporary file in the directory `dir`.
+proc createTempFile*(prefix, suffix: string, dir = ""): tuple[cfile: File, path: string] =
+  ## Creates a new temporary file in the directory `dir`, which must exist
+  ## (empty `dir` will resolve to `getTempDir()`).
   ## 
   ## This generates a path name using `genTempPath(prefix, suffix, dir)` and
   ## returns a file handle to an open file and the path of that file, possibly after
   ## retrying to ensure it doesn't already exist.
   ## 
-  ## If failing to create a temporary file, `IOError` will be raised.
+  ## If failing to create a temporary file, `OSError` will be raised.
   ##
   ## .. note:: It is the caller's responsibility to remove the file when no longer needed.
+  runnableExamples:
+    import std/[os, nre]
+    doAssertRaises(OSError): discard createTempFile("", "", "nonexistent")
+    let (cfile, path) = createTempFile("D20210501T170028", "end")
+    cfile.write "foo"
+    cfile.setFilePos 0
+    assert path.lastPathPart.contains(re"^D20210501T170028(\w+)end$")
+    assert readAll(cfile) == "foo"
+    close cfile
+    assert readFile(path) == "foo"
+    removeFile(path)
+  # xxx why does above work without `cfile.flushFile` ?
   let dir = getTempDirImpl(dir)
-  createDir(dir)
```

Comment:
why?

Reply:
separation of concerns; user should create the dir (and in most cases they'd use the default dir which shouldn't have to be created)

Reply:
Well, you need to make sure the system directories exits. `getTempDir` won't check it.

```nim
proc getTempDirImpl(dir: string): string {.inline.} =
  result = dir
  if result.len == 0:
    result = getTempDir()
```

Reply:
that should be handled by the OS though; in which case is `getTempDir()` returning a non-existent path?

eg on osx it returns something like: `/var/folders/ll/<randomstring>/T`, which exists (and is relied upon by other apps)

Reply:
I agree. If the system's temporary directory doesn't exist, then the user should be the one to fix that.